kotsadm/pkg/handlers/troubleshoot.go

```diff
@@ -99,6 +119,81 @@ func GetSupportBundleFiles(w http.ResponseWriter, r *http.Request) {
 	JSON(w, 200, getSupportBundleFilesResponse)
 }
 
+func ListSupportBundles(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Headers", "content-type, origin, accept, authorization")
+
+	if r.Method == "OPTIONS" {
+		w.WriteHeader(200)
+		return
+	}
+
+	sess, err := session.Parse(r.Header.Get("Authorization"))
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(401)
+		return
+	}
+
+	// we don't currently have roles, all valid tokens are valid sessions
+	if sess == nil || sess.ID == "" {
+		logger.Error(err)
+		w.WriteHeader(401)
+		return
+	}
+
+	appSlug := mux.Vars(r)["appSlug"]
+
+	a, err := app.GetFromSlug(appSlug)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(500)
+		return
+	}
+
+	supportBundles, err := supportbundle.List(a.ID)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(500)
+		return
+	}
+
+	responseSupportBundles := []ResponseSupportBundle{}
+	for _, bundle := range supportBundles {
+		licenseType, err := supportbundle.GetLicenseType(bundle.ID)
+		if err != nil {
+			logger.Error(errors.Wrapf(err, "failed to get license type for bundle %s", bundle.Slug))
```

Comment:
missing return

Reply:
hmmm, i don't think failing to get the license type should fail the whole request or not include the support bundle in the list